GTFS.DB.PostgreSQL/Collections/PostgreSQLAgencyCollection.cs

```diff
@@ -0,0 +1,236 @@
+            using (var command = _connection.CreateCommand())
+            {
+                command.CommandText = sql;
+                command.Parameters.Add(new NpgsqlParameter(@"feed_id", DbType.Int64));
+                command.Parameters.Add(new NpgsqlParameter(@"id", DbType.String));
+                command.Parameters.Add(new NpgsqlParameter(@"agency_name", DbType.String));
+                command.Parameters.Add(new NpgsqlParameter(@"agency_url", DbType.String));
+                command.Parameters.Add(new NpgsqlParameter(@"agency_timezone", DbType.String));
+                command.Parameters.Add(new NpgsqlParameter(@"agency_lang", DbType.String));
+                command.Parameters.Add(new NpgsqlParameter(@"agency_phone", DbType.String));
+                command.Parameters.Add(new NpgsqlParameter(@"agency_fare_url", DbType.String));
+
+                command.Parameters[0].Value = _id;
+                command.Parameters[1].Value = entity.Id;
+                command.Parameters[2].Value = entity.Name;
+                command.Parameters[3].Value = entity.URL;
+                command.Parameters[4].Value = entity.Timezone;
+                command.Parameters[5].Value = entity.LanguageCode;
+                command.Parameters[6].Value = entity.Phone;
+                command.Parameters[7].Value = entity.FareURL;
+
+                command.Parameters.Where(x => x.Value == null).ToList().ForEach(x => x.Value = DBNull.Value);
+                command.ExecuteNonQuery();
+            }
+        }
+
+
+        public void AddRange(IUniqueEntityCollection<Agency> entities)
+        {
+            using (var writer = _connection.BeginBinaryImport("COPY agency (feed_id, id, agency_name, agency_url, agency_timezone, agency_lang, agency_phone, agency_fare_url) FROM STDIN (FORMAT BINARY)"))
+            {
+                foreach (var agency in entities)
+                {
+                    writer.StartRow();
+                    writer.Write(_id, NpgsqlTypes.NpgsqlDbType.Integer);
+                    writer.Write(agency.Id, NpgsqlTypes.NpgsqlDbType.Text);
+                    writer.Write(agency.Name, NpgsqlTypes.NpgsqlDbType.Text);
+                    writer.Write(agency.URL, NpgsqlTypes.NpgsqlDbType.Text);
+                    writer.Write(agency.Timezone, NpgsqlTypes.NpgsqlDbType.Text);
+                    writer.Write(agency.LanguageCode, NpgsqlTypes.NpgsqlDbType.Text);
+                    writer.Write(agency.Phone, NpgsqlTypes.NpgsqlDbType.Text);
+                    writer.Write(agency.FareURL, NpgsqlTypes.NpgsqlDbType.Text);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the entity with the given id.
+        /// </summary>
+        /// <param name="entityId"></param>
+        /// <returns></returns>
+        public Agency Get(string entityId)
+        {
+            throw new NotImplementedException();
```

Comment:
maybe implement this one

Reply:
They have never been implemented though. Not sure we would ever use it

Reply:
yeah, just realised that they're not necessary for the LINQ statements that we use